### ISSUE TITLE
Consider parallel ways in Bicycle quest

### ISSUE THREAD

**Issue by naomap** [westnordost/StreetComplete#1250]
Hello.
The bicycle quest apparently only considers each way individually. However in many cases cycleways or shared busways are mapped as a separate way. This results in users creating wrong data, such as adding a cycle track where it as already been mapped.
Example with this share_busway : https://www.openstreetmap.org/changeset/63789825
Is it possible to improve the quest to consider parallel ways ?
Regards,
Antoine.

**Comment by matkoniecz** [comment]
image of the street in street view: https://www.google.pl/maps/@48.8646259,2.3553856,3a,75y,20.25h,79.17t/data=!3m5!1e1!3m3!1sqlJCEGY7sQ_coObV8XI25g!2e0!6s%2F%2Fgeo2.ggpht.com%2Fcbk%3Fpanoid%3DqlJCEGY7sQ_coObV8XI25g%26output%3Dthumbnail%26cb_client%3Dmaps_sv.tactile.gps%26thumb%3D2%26w%3D203%26h%3D100%26yaw%3D82.73496%26pitch%3D0%26thumbfov%3D100

**Comment by naomap** [comment]
You didn't get my point. Yes there is a busway, but it is ALREADY mapped as https://www.openstreetmap.org/way/55129128
So adding the tag cycleway:right to https://www.openstreetmap.org/way/55129129 is wrong.
In fact the quest should NOT ask whether there is a cycleway or shared busway on the right in this case, as it is already mapped as a separate way.

**Comment by matkoniecz** [comment]
> You didn't get my point.

Why you thought so? How I should avoid such misunderstanding in a future? I found photos of the affected location (to check whatever current tagging is correct), noticed that it is viable to tag is this way. I just left it as a note to save time of others.

I added nothing more as I had no additional useful to post.

**Comment by naomap** [comment]
The problem in that sort of case (this is not the only occurrence of course), is that StreetComplete creates redundant data. In fact no, this is not redundant data but wrong data. Having an highway with a tag cycleway:right=share_busway PLUS a parallel way on its right carrying the tags psv=yes and bicyle=yes means that there are two parallel cycle routes whereas there is only one !

**Comment by westnordost** [comment]
I think the problem here is not StreetComplete, it was simply a wrong entry made by that user. He should have answered that there is no cycleway. The map is displayed in StreetComplete, both the bus lane part of the road and the other road.

The map looks like this:
![bildschirmfoto 2018-11-01 um 12 24 02](https://user-images.githubusercontent.com/4661658/47849199-0fee9f00-ddd1-11e8-8e2e-6fb962df7282.png)

However, one can see that it is not very clear that there are two roads parallel to each other displayed here (current version of StreetComplete doesn't even show the bus lane with a dashed outline) because the roads melt into each other just like the Mapnik map on openstreetmap.org on zoom level 18 and below.

I venture to say that two roads are so close to each other is somewhat of an exception (in reality, it **is** one road, only it is not mapped that was in OSM), but still, do you have a suggestion how to make it more clear to users of the app that there are two ways and the user should only answer the question for the way highlighted?

**Comment by matkoniecz** [comment]
>  do you have a suggestion how to make it more clear to users of the app that there are two ways and the user should only answer the question for the way highlighted?

Allow higher zoom in the map style? Make roads in map style narrower to reduce merging problem?

**Comment by westnordost** [comment]
Going into this direction, maybe derive the final rendered width in meters (at high zoom levels) from `width` or `lanes`.

**Comment by westnordost** [comment]
> This is not widely used yet, however maybe SC could check whether the way holds this tag before requesting the user whether there is some cycleway. Maybe it does already ?

Yes, it does. In fact, if the road is tagged with any cycleway value, the cycleway is not asked. See here for the whole Overpass query:

https://github.com/westnordost/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/bikeway/AddCycleway.java#L189-L219
